### PR TITLE
Feature/uuids and dtos

### PR DESCRIPTION
**Pull request by Sparky-Kitty**

### CODE REVIEW

**Comment by Sparky-Kitty** [comment]
Still having issues with "unable to find matching" result that does exist, but we are close!

File: src/services/service-slots/service-slots.service.ts [hunk 1]
```diff
@@ -58,8 +59,9 @@ export class ServiceSlotsService {
         return await this.serviceSlotRepository.save(serviceSlot);
     }
 
-    async updateServiceSlotState(id: number, state: API.ServiceSlotStates): Promise<ServiceSlot> {
+    async updateServiceSlotState(slotUuid: string, state: API.ServiceSlotStates): Promise<ServiceSlot> {
         // Check the validity of the state before proceeding
+        console.log(slotUuid);
```
Comment (d4search) [inline src/services/service-slots/service-slots.service.ts:64]: Remove console log.

**Review by d4search — CHANGES_REQUESTED** [review]

File: src/services/service-slots/service-slots.service.ts [hunk 2]
```diff
@@ -88,7 +87,7 @@ export class ServiceSlotsService {
 
             if (state === API.ServiceSlotStates.Accepted) {
                 const acceptedSlotsCount = await slotQueryBuilder
-                    .where('service_slot.service_id = :serviceId', { serviceId: slot.service.id })
+                    .where('service.uuid = :serviceId', { serviceId: slot.service?.uuid })
```
Comment (d4search) [inline src/services/service-slots/service-slots.service.ts:90]: How can the service not have a uuid?

File: src/services/service-slots/service-slots.service.ts [hunk 3]
```diff
@@ -104,7 +103,7 @@ export class ServiceSlotsService {
                 await slotQueryBuilder
                     .update()
                     .set({ state: API.ServiceSlotStates.Rejected })
-                    .where('service_slot.client_user_id = :userId', { userId: slot.client.id })
+                    .where('client_user_id = :clientId', { clientId: slot.client?.id })
```
Comment (d4search) [inline src/services/service-slots/service-slots.service.ts:106]: How can the client not have an id?

**Pull request merged**